src/components/Patients/PatientEdit.js

```diff
@@ -0,0 +1,259 @@
+    ),
+  firstName: Yup.string()
+    .max(15, "Must be 15 characters or less")
+    .required("Required"),
+  lastName: Yup.string()
+    .max(20, "Must be 20 characters or less")
+    .required("Required"),
+  dateOfBirth: Yup.date()
+    .nullable()
+    .required("Required"),
+  address: Yup.string()
+    .max(15, "Must be 20 characters or less")
+    .required("Required"),
+  city: Yup.string()
+    .max(15, "Must be 15 characters or less")
+    .required("Required"),
+  phoneNumber: Yup.string()
+    .matches(/^[+]*[-\s/0-9]*$/)
+    .required("Required"),
+  email: Yup.string().email("Invalid email address").required("Required"),
+});
+
+const PatientEdit = () => {
+  const navigate = useNavigate();
+  const { slug } = useParams();
+  const [patient, setPatient] = useState();
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState();
+
+  useEffect(() => {
+    const fetch = async () => {
+      try {
+        setLoading(true);
+        const response = await getPatientBySlug(slug);
+        setPatient(response);
+      } catch (e) {
+        setError("Unable to fetch patient");
+      } finally {
+        setLoading(false);
+      }
+    };
+    fetch();
+  }, [slug]);
+
+  const handleFileUpload = async (e, setFieldValue) => {
+    const file = e.target.files[0];
+
+    if (file) {
+      const base64 = await convertToBase64(file);
+      setFieldValue("image", base64);
+    }
+  };
+
+  const handleSubmit = async (values, { setSubmitting }) => {
+    try {
+      await editPatient({
+        slug,
+        image: values.image,
+        firstName: values.firstName,
+        lastName: values.lastName,
+        dateOfBirth: values.dateOfBirth,
+        address: values.address,
+        city: values.city,
+        phoneNumber: values.phoneNumber,
+        email: values.email,
+      });
+      toast.success("Patient updated!");
+      navigate("/patients/");
+    } catch (err) {
+      toast.error("Unable to update patient!");
+    } finally {
+      setSubmitting(false);
+    }
+  };
+
+  return (
+    <StyledContainer>
+      <StyledHeader>Edit patient</StyledHeader>
+      <Breadcrumb>
+        <Breadcrumb.Section link><Link to="/patients">Patients</Link></Breadcrumb.Section>
+        <Breadcrumb.Divider />
+        <Breadcrumb.Section link><Link to={`/patients/${slug}`}>Patient details</Link></Breadcrumb.Section>
+        <Breadcrumb.Divider />
+        <Breadcrumb.Section active>Edit patient</Breadcrumb.Section>
+      </Breadcrumb>
+      <Divider />
+      {patient
+      && (
+      <Formik
+        initialValues={{ ...patient, dateOfBirth: Moment(patient.dateOfBirth).format("YYYY-MM-DD"), image: encodeURI(Buffer.from(patient.image.data)) }}
+        validationSchema={validationSchema}
+        onSubmit={handleSubmit}
+      >
+
+        {({
+          values, touched, errors, setFieldValue,
+        }) => (
+          <Form>
+            <StyledTopContainer>
+              <ImageContainer>
```

Comment:
Is there a chance you can create a component to image upload - same as `InputField` and call it `ImageField` or `AvatarField`?